modules/testkit/shared/src/main/scala/gem/arb/ArbTwilightBoundedNight.scala

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package arb
+
+import gsp.math.skycalc.TwilightBoundType
+import gem.math.{ ObservingNight, TwilightBoundedNight }
+
+import org.scalacheck._
+import org.scalacheck.Arbitrary._
+
+trait ArbTwilightBoundedNight {
+
+  import ArbObservingNight._
+  import gsp.math.arb.ArbTwilightBoundType._
+
+  implicit val arbTwilightBoundedNight: Arbitrary[TwilightBoundedNight] =
+    Arbitrary {
+      (
+        for {
+          b <- arbitrary[TwilightBoundType]
+          n <- arbitrary[ObservingNight]
+        } yield n.twilightBounded(b)
+      ).suchThat(_.isDefined).map(_.get)
```

Comment:
I haven't seen this before. will it make ignore non valid generated values?

Reply:
Yes. This is because all safe `TwilightBoundedNight`s return `Option` in case there are no twilight bounds for the specified parameters.

In our code we are supposed to only be calling with the current `Site`s, where we know we always have twilight bounds, so again, we could use unsafe get. We could even do unsafe gets in all constructors? IDK the right decision here, is this code ever going to be used for other places on Earth?

Reply:
That sounds fine, try to
Move the option up as much as posssible though I doubt an arbitrary of option is very useful